compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4963,7 +4986,13 @@ private STNode parseWhileStatement() {
         STNode condition = parseExpression();
         STNode whileBody = parseBlockNode();
         endContext();
-        return STNodeFactory.createWhileStatementNode(whileKeyword, condition, whileBody);
+        STNode onFailClause;
+        if (peek().kind == SyntaxKind.ON_KEYWORD) {
```

Comment:
Need to update the doc comment with the new grammar. 
Applicable for all places.

Reply:
Fixed with 3189bc6c6a9ee435ca57fa863f10d8e4f357ab6b

Reply:
Fixed with 3189bc6c6a9ee435ca57fa863f10d8e4f357ab6b